tests/Unit/ProtoTests/Basic/out/src/Client/BasicClient.php

```diff
@@ -208,4 +211,30 @@ public function methodWithArgs(RequestWithArgs $request, array $callOptions = []
     {
         return $this->startApiCall('MethodWithArgs', $request, $callOptions)->wait();
     }
+
+    /** Configure the gapic configuration to use a service emulator. */
+    private function setEmulatorConfig()
+    {
+        $emulatorHost = getenv('BASIC_EMULATOR_HOST');
+        if (!empty($emulatorHost)) {
+            if ($scheme = parse_url($emulatorHost, PHP_URL_SCHEME)) {
+                $search = $scheme . '://';
+                $emulatorHost = str_replace($search, '', $emulatorHost);
```

Comment:
This logic doesn't make sense to me - if someone provides a URL scheme in `BASIC_EMULATOR_HOST`, we strip it out? Why?

Reply:
I have reused the logic from this core trait: https://github.com/googleapis/google-cloud-php/blob/9b40c2cc7e67d101ca0432a29c613b7c48053b4f/Core/src/EmulatorTrait.php#L31

Reply:
SGTM